src/pages/signup/components/SignupForm/index.tsx

```diff
@@ -0,0 +1,123 @@
+import styles from './SignupForm.module.scss';
+import { useRouter } from 'next/router';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import axios from '@/apis/axios';
+import { isAxiosError } from 'axios';
+import PageButton from '@/components/Button/PageButton';
+import Input from '@/components/Inputs/Input';
+import setToast from '@/utils/setToast';
+
+type FormValues = {
+  email: string;
+  nickname: string;
+  password: string;
+  passwordConfirm: string;
+};
+
+const schema = z
+  .object({
+    email: z
+      .string()
+      .trim()
+      .min(1, { message: '이메일을 입력해 주세요.' })
+      .email({ message: '이메일 형식으로 작성해 주세요.' }),
+    nickname: z
+      .string()
+      .trim()
+      .min(1, { message: '닉네임을 입력해 주세요.' })
+      .max(10, { message: '열 자 이하로 작성해주세요.' }),
+    password: z
+      .string()
+      .trim()
+      .min(1, { message: '비밀번호를 입력해 주세요.' })
+      .min(8, { message: '8자 이상 입력해주세요.' }),
+    passwordConfirm: z.string().trim().min(1, '비밀번호를 입력해 주세요.'),
+  })
+  .refine((data) => data.password === data.passwordConfirm, {
+    path: ['passwordConfirm'],
+    message: '비밀번호가 일치하지 않습니다.',
+  });
+
+function SignupForm() {
+  const router = useRouter();
+  const {
+    register,
+    formState: { errors, isValid },
+    handleSubmit,
+    setError,
+    reset,
+  } = useForm<FormValues>({
+    resolver: zodResolver(schema),
+    mode: 'all',
+    defaultValues: {
+      email: '',
+      nickname: '',
+      password: '',
+      passwordConfirm: '',
+    },
+  });
+
+  const handleValidSubmit = async (data: FormValues) => {
+    try {
+      await axios.post('/users', data);
+      setToast('success', '가입이 완료되었습니다.');
+      router.push('/login');
+    } catch (e) {
+      if (isAxiosError(e)) {
+        const message =
+          e.response?.data.message || '다른 이메일을 입력해 주세요.';
+        setError('email', { message });
+        return;
+      }
+      setToast('warn', '다시 시도해 주세요.');
+      reset();
```

Comment:
```suggestion
        setError('email', { message });
        return;
      }
      setToast('warn', '다시 시도해 주세요.');
      reset();
```
`setError`로 어떤 이유로 회원가입이 안되는지 알려주니깐 토스트까지 사용할 필요가 없을 것 같아요!
`reset `은 인풋을 지워주는거? 같은데 회원가입 실패하면 모든 인풋에 값이 사라지는건가요??

Reply:
> `setError`로 어떤 이유로 회원가입이 안되는지 알려주니깐 토스트까지 사용할 필요가 없을 것 같아요! `reset `은 인풋을 지워주는거? 같은데 회원가입 실패하면 모든 인풋에 값이 사라지는건가요??

아래 코드는 `axios` error가 아닌 경우일 때의 에러 처리인데 현재 상태에선 `axios` 말고는 catch 문으로 넘어갈 일이 없긴 하지만 있으면 좋을 거 같아서 작성했습니다!
`reset()`은 모든 인풋이 사라지는 거 맞아요! 아래로 넘어왔다는 거는 특정 `Input`의 에러를 특정할 수 없어서 회원가입 버튼이 계속 활성화 되어있고 그럴 경우 버튼을 계속 누를 수 있기 때문에 초기화를 진행하는 게 좋을 거 같다고 판단했습니다.
```ts
setToast('warn', '다시 시도해 주세요.');
reset();
```

Reply:
> axios error가 아닌 경우?가 어떤 말일까요??

지금 코드로는 axios 에러 말고는 특별히 에러 날 상황이 없긴 하지만 catch문에 axios 에러를 if문으로 에러 핸들링을 해주고 있는데 저희가 모르는 어떤 에러가 try에서 났을 경우를 대비해서 작성했어요!